---
title: 'Tutorial: process_radtags'
output:
  html_document: default
  word_document: default
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, eval = FALSE,
                      echo=TRUE, warning=FALSE, message=FALSE,
                      tidy = TRUE, collapse = TRUE,
                      results = 'hold')
```
\
\

### Timings 
  1. Run process_radtags (ca. 3 hours)
  
All done in EVE cluster
    
\
\

### 1. Run process_ radtags

This is all done in the EVE cluster, it will take some time and in the meantime we do other things, but at least it gives you a feel for how you go from sequencer files to individual files

```{r}
# process_radtags####

#!/bin/bash                                                                                                                            
#SBATCH --job-name=process_radtags
#SBATCH --mail-user=christopher_david.barratt@uni-leipzig.de
#SBATCH --mail-type=BEGIN,END,FAIL,TIME_LIMIT
#SBATCH --chdir /data/%u  
#SBATCH --output=/work/%u/Workshop_3/job_logs/%x-%j.log
#SBATCH --cpus-per-task=1 
#SBATCH --mem-per-cpu=8G
#SBATCH --time=168:00:00
 

## Load modules and activate software

module purge
module load Anaconda3
conda activate /gpfs0/global/apps/stacks_2.61
 
# process_radtags - take raw data from sequencer and output sequences per individual using barcodes and restriction enzyme information
# -p = file path of raw data from sequencer 
# -o = output file path of where you want your individual data to go 
# -b = barcode identifier (one file showing all barcodes for each individual) 
# --inline_inline: barcode is inline with sequence, occurs on single and paired-end read.
# --renz_1 = restriction enzyme used
# --renz_2 = restriction enzyme used
# -c,--clean = clean data, remove any read with an uncalled base.
# -q,--quality = discard reads with low quality scores.
# --filter_illumina = discard reads that have been marked by Illumina's chastity/purity filter as failing.
# -s = set the score limit. If the average score within the sliding window drops below this value, the read is discarded (default 10).i
# -i = input file type, either 'fastq', 'gzfastq' (gzipped fastq), 'bam', or 'bustard' (default: guess, or gzfastq if unable to).
# -y = output type, either 'fastq', 'gzfastq', 'fasta', or 'gzfasta' (default: match input type).
# -t = truncate final read length to this value.

process_radtags -1 /data/life_on_the_edge/Workshop_3/Myotis_escalarai/raw/BAT_01_NoIndex_L008_R1_001.fastq.gz -2 /data/life_on_the_edge/Workshop_3/Myotis_escalarai/raw/BAT_01_NoIndex_L008_R2_001.fastq.gz -o /data/life_on_the_edge/Workshop_3/Myotis_escalarai/demultiplexed/ -b /data/life_on_the_edge/Workshop_3/Myotis_escalarai/barcodes/barcodes_1 --inline_inline --renz_1 sbfI --renz_2 nlaIII -c -q -E phred33 -i gzfastq -y gzfastq --filter_illumina -s 20 -t 90 

# You can try it yourself now for barcodes_2, barcodes_3 etc.
# try changing the restriction enzyme
# try changing s and t parameters
# see what happens when getting the inline-inline wrong..

process_radtags -1 /data/life_on_the_edge/Workshop_3/Myotis_escalarai/raw/BAT_01_NoIndex_L008_R1_001.fastq.gz -2 /data/life_on_the_edge/Workshop_3/Myotis_escalarai/raw/BAT_01_NoIndex_L008_R2_001.fastq.gz -o /data/life_on_the_edge/Workshop_3/Myotis_escalarai/demultiplexed/ -b /data/life_on_the_edge/Workshop_3/Myotis_escalarai/barcodes/barcodes_1 --inline_inline --renz_1 sbfI --renz_2 nlaIII -c -q -E phred33 -i gzfastq -y gzfastq --filter_illumina -s 20 -t 90 

```

