---
title: 'Tutorial: denovo_map'
output:
  html_document: default
  word_document: default
---

```{r global_options, include=FALSE}
knitr::opts_chunk$set(fig.width=12, fig.height=8, eval = FALSE,
                      echo=TRUE, warning=FALSE, message=FALSE,
                      tidy = TRUE, collapse = TRUE,
                      results = 'hold')
```
\
\

### Timings 
  1. Run denovo_map (2 hours?)
  1. Run VCFtools to assess output file quality (10 min)

All steps done in EVE cluster  
  
\
\

### Objectives

This exercise will show you how to run denovo_map from start to finish (assuming that you have demultiplexed data after running process_radtags). For this exercise we will use some already processed data for Hyphaene coriacea, a palm species from Madagascar. We'll then check the quality of the output files using VCFtools.

\
\

### 1. Run denovo_map

This is all done in the EVE cluster, you can make your own scripts to do this or use the one provided (run_Admixture.sh). It can be submitted via sbatch in the EVE cluster, but be sure to modify the email address and output paths to your own details

```{r}
#!/bin/bash                                                                                                                            
#SBATCH --job-name=denovo_map
#SBATCH --mail-user=christopher_david.barratt@uni-leipzig.de
#SBATCH --mail-type=BEGIN,END,FAIL,TIME_LIMIT
#SBATCH --output=~/work/ddRAD-seq_workshop/job_logs/%x-%j.log
#SBATCH --cpus-per-task=4 
#SBATCH --mem-per-cpu=50G
#SBATCH --time=48:00:00

LAURA's CLEANED CODE
```

\
\

### 2. Assess outputs using VCFtools

We can assess the quality of reads (coverage, % missing data per sample etc.) to see which individuals have been sequenced well, and which ones have not. This would be important for later analyses making sure that you have as many homologous reads (i.e. loci and SNPs) as possible for your given questions, even if you lose a few individuals.

```{r}
#!/bin/bash                                                                                                                            
#SBATCH --job-name=vcftools
#SBATCH --mail-user=christopher_david.barratt@uni-leipzig.de
#SBATCH --mail-type=BEGIN,END,FAIL,TIME_LIMIT
#SBATCH --output=~/work/ddRAD-seq_workshop/job_logs/%x-%j.log
#SBATCH --cpus-per-task=4 
#SBATCH --mem-per-cpu=50G
#SBATCH --time=48:00:00

LAURA's CLEANED CODE
```



