<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tutorial: Admixture</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ddRAD-seq_workshop</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00_setup.html">00 - Setting up the data and scripts for analysis</a>
    </li>
    <li>
      <a href="01_download_data.html">01 - Downloading data from ENA</a>
    </li>
    <li>
      <a href="02_process_radtags.html">02 - Demultiplexing sequencing data (process_radtags)</a>
    </li>
    <li>
      <a href="03_denovo_map.html">03 - Running Stacks 2 (denovo_map)</a>
    </li>
    <li>
      <a href="04_Admixture.html">04 - Population structure (Admixture)</a>
    </li>
    <li>
      <a href="05_sNMF.html">05 - Population structure (sNMF)</a>
    </li>
    <li>
      <a href="06_DAPC.html">06 - Population structure (DAPC)</a>
    </li>
    <li>
      <a href="07_RAxML.html">07 - Phylogenetic relationships (RAxML)</a>
    </li>
    <li>
      <a href="08_F-stats.html">08 - Genetic Diversity (F-stats)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/cd-barratt/ddRAD-seq_workshop/">GitHub</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Tutorial: Admixture</h1>

</div>


<p><br />
<br />
</p>
<div id="timings" class="section level3">
<h3>Timings</h3>
<ol style="list-style-type: decimal">
<li>Run admixture (in EVE cluster, ~3 min) and then copy results to your local machine</li>
<li>Load data (&lt; 1 min)</li>
<li>Plot CV scores (&lt; 1 min)</li>
<li>Make barplot of Admixture results (&lt; 1 min)</li>
</ol>
<p>Step 1 done in EVE cluster, Steps 2-4 done locally on your machines</p>
<p><br />
<br />
</p>
</div>
<div id="objectives" class="section level3">
<h3>Objectives</h3>
<p>For this exercise and the remainder of the workshop we will use some published amphibian data to look at population structure, phylogeny and genetic diversity estimates. The data are for <a href="https://amphibiaweb.org/species/3649"><em>Leptopelis flavomaculatus</em></a>, a forest tree frog from Tanzania, and were published <a href="https://onlinelibrary.wiley.com/doi/full/10.1111/mec.14862">here</a>.</p>
<p><br />
</p>
<p>For this exercise in particular using <a href="https://dalexander.github.io/admixture/publications.html">Admixture</a> software, we will learn how to go from a Stacks output file to investigate population structure, find statistical support to show the ‘best’ explained population structure (i.e. number of clusters), and then plot a nice barplot to show population assignment for each individual.</p>
<p><br />
<br />
</p>
</div>
<div id="run-admixture-on-the-eve-cluster" class="section level3">
<h3>1. Run Admixture (on the EVE cluster)</h3>
<p>#This is all done in the EVE cluster, you can make your own scripts to do this or use the one provided (04_Admixture.sh). It can be submitted via sbatch in the EVE cluster, but be sure to modify the email address and output paths to your own details.</p>
<pre class="r"><code>#!/bin/bash                                                                                                                            

#SBATCH --job-name=Admixture
#SBATCH --mail-user=christopher_david.barratt@uni-leipzig.de
#SBATCH --mail-type=BEGIN,END,FAIL,TIME_LIMIT
#SBATCH --output=/work/$USER/ddRAD-seq_workshop/job_logs/%x-%j.log
#SBATCH --cpus-per-task=1
#SBATCH --mem-per-cpu=15G
#SBATCH --time=2:00:00

 
#load environment
module purge
module load Anaconda3
source activate /global/apps/plink_1.90


# Converting ped and map files to bed format for Admixture
plink --file /work/$USER/ddRAD-seq_workshop/data/Exercises_4-8/Admixture/Lflav --make-bed --recode --out /work/$USER/ddRAD-seq_workshop/outputs/Exercise_4/

#run admixture
cd /work/$USER/ddRAD-seq_workshop/outputs/Exercise_4/
bash
source activate /gpfs0/global/apps/admixture_1.3.0
for K in 1 2 3 4 5 6 7 8 9 10; \
do admixture --cv=10 ./Lflav.bed $K| tee log_Lflav.${K}.out; done

# grep to combine all results
grep -h CV log_Lflav*.out &gt; ./RESULTS_Leptopelis_flavomaculatus_admixture.txt

#sed to clean results for later analysis
sed -e &#39;s/CV error (K=//g&#39; ./RESULTS_Leptopelis_flavomaculatus_admixture.txt | sed &#39;s/)://g&#39; &gt; ./RESULTS_Leptopelis_flavomaculatus_admixture_clean.txt

# replace all single digit numbers to begin with 0s for numerical sorting
sed -e &#39;s/1 /01 /g&#39; ./RESULTS_Leptopelis_flavomaculatus_admixture_clean.txt \
| sed -e &#39;s/2 /02 /g&#39; \
| sed -e &#39;s/3 /03 /g&#39; \
| sed -e &#39;s/4 /04 /g&#39; \
| sed -e &#39;s/5 /05 /g&#39; \
| sed -e &#39;s/6 /06 /g&#39; \
| sed -e &#39;s/7 /07 /g&#39; \
| sed -e &#39;s/8 /08 /g&#39; \
| sed -e &#39;s/9 /09 /g&#39;&gt; ./tmp.txt\

# sort output file (01-10) and remove old temporary one
sort ./tmp.txt &gt; ./RESULTS_Leptopelis_flavomaculatus_admixture_clean.txt
rm -rf ./tmp.txt</code></pre>
<p><br />
</p>
<p>When the Admixture job is complete we need to copy the output files to our desktop. We can use Cyberduck/FileZilla for this (I’ll show you), or we can use <a href="https://wiki.ufz.de/eve/index.php/File_Transfer">rsync</a>. To use rsync open up another version of your Terminal/PuTTy and type the following (edited to match your account details and local desktop path):</p>
<pre class="r"><code># transfer from the cluster
rsync -avhP -e &#39;ssh -p 8022&#39; \
  barratt@localhost:/work/$USER/ddRAD-seq_workshop \
  /Users/cb76kecu/Desktop</code></pre>
<p><br />
</p>
</div>
<div id="load-data" class="section level3">
<h3>2. Load data</h3>
<p>Load the cleaned Admixture output file that we will analyse. This is a simple text file with the number of clusters (K) and the corresponding 10-fold cross-validation(CV) score. Be aware here and further on that directory separators are different between mac (/) and Windows (), so check you are using the correct ones if you are having trouble reading files</p>
<pre class="r"><code>results &lt;- read.delim(&quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/outputs/Exercise_4/RESULTS_Leptopelis_flavomaculatus_admixture_clean.txt&quot;,
    sep = &quot; &quot;, header = FALSE)</code></pre>
<p><br />
<br />
</p>
</div>
<div id="plot-cv-scores" class="section level3">
<h3>3. Plot CV scores</h3>
<p>We will first set the plot margins and then set an output file (png) with specific sizes and resolutions Then we plot the results (first points, then a line to join the points)</p>
<pre class="r"><code>par(mar = c(2, 2, 2, 2))
png(filename = &quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/outputs/Exercise_4/Admixture_Results_CV_scores.png&quot;,
    type = &quot;quartz&quot;, height = 8, width = 8, units = &quot;in&quot;, res = 300)
plot(results, pch = 19, col = &quot;black&quot;, cex = 0.25, xlab = &quot;k clusters&quot;, ylab = &quot;CV score&quot;,
    main = &quot;Cross validation scores across k (Admixture&quot;)
lines(results, lty = 1, lwd = 1.5, col = &quot;steelblue2&quot;)
dev.off()</code></pre>
<p>Because this dataset is for relatively highly structured (low-dispersal) amphibians, the ‘elbow’ in the plot is very clear - with a lower CV score at K=4. This is not always the case for all datasets, so other approaches might be useful to use a few different methods to look at population structure (reviewers anyway generally like this approach).</p>
<p><br />
<br />
</p>
</div>
<div id="make-barplot-of-admixture-results" class="section level3">
<h3>4. Make barplot of Admixture results</h3>
<p>Now that we know the optimal value of K we can plot the Admixture coefficients across all of our samples (n=59 in this case). We’ll read the Lflav.4.Q output file from Admixture and order it to make it look nicer. Each .Q file is a prediction of population membership per individual across the given number of clusters you decieded to test when running Admixture (i.e. 1:10). In the case of Lflav.4.Q It is a matrix of 59 rows (one per individual) and 4 columns (i.e. K=4). The probability of membership to each population cluster is shown for each individual. We’ll then plot it, add individual labels for each sample and then save the final plot.</p>
<pre class="r"><code># plot bars
tbl = read.table(&quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/outputs/Exercise_4/Lflav.4.Q&quot;)
ord = tbl[order(tbl$V1, tbl$V3, tbl$V2, tbl$V4), ]
png(filename = &quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/outputs/Exercise_4/Admixture_Results_Barplot.png&quot;,
    type = &quot;quartz&quot;, height = 8, width = 30, units = &quot;in&quot;, res = 300)
labels &lt;- read.csv(&quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/data/Exercises_4-8/Admixture/pop_labels.csv&quot;,
    header = FALSE)
labels &lt;- data.frame(labels[, 1])
ord_with_labels &lt;- data.frame(ord[, 1:4], row.names = labels[, 1])
par(mar = c(20, 4, 1, 1))
bp = barplot(t(as.matrix(ord_with_labels)), space = (0.04), col = c(&quot;blue&quot;, &quot;turquoise&quot;,
    &quot;darkblue&quot;, &quot;lightblue&quot;), xlab = &quot;&quot;, ylab = &quot;Ancestry&quot;, border = 1, las = 2,
    cex.names = 0.9)
dev.off()
</code></pre>
<p><br />
<br />
</p>
</div>
<div id="on-your-own" class="section level3">
<h3>On your own…</h3>
<p>If you want, try plotting k=1 - k=5, feel free to change colour schemes and the order of the tbl object. You can also edit the CV validation method t be –cv=5 for example when running Admixture. If you have your own data you can try and use this script to run Admixture and then plot your own outputs</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
