<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tutorial: Discriminant Analysis of Principal Components (DAPC)</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ddRAD-seq_workshop</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01a_download_data.html">01a - Downloading data from ENA</a>
    </li>
    <li>
      <a href="01b_process_radtags.html">01b - Demultiplexing sequencing data (process_radtags)</a>
    </li>
    <li>
      <a href="02_denovo_map.html">02 - Running Stacks 2 (denovo_map)</a>
    </li>
    <li>
      <a href="03_Admixture.html">03 - Population structure (Admixture)</a>
    </li>
    <li>
      <a href="04_sNMF.html">04 - Population structure (sNMF)</a>
    </li>
    <li>
      <a href="05_DAPC.html">05 - Population structure (DAPC)</a>
    </li>
    <li>
      <a href="06_RAxML.html">06 - Phylogentic relationships (RAxML)</a>
    </li>
    <li>
      <a href="07_Fstats.html">07 - Genetic Diversity)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/cd-barratt/ddRAD-seq_workshop/">GitHub</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Tutorial: Discriminant Analysis of Principal Components (DAPC)</h1>

</div>


<p><br />
<br />
</p>
<div id="timings" class="section level3">
<h3>Timings</h3>
<ol style="list-style-type: decimal">
<li>Initial setup and package install (&lt;1 min)</li>
<li>Load data abd convert to genind format for Adegenet (&lt; 1 min)</li>
<li>Run the DAPC and plot the BIC scores (3 mins)</li>
<li>Make barplot of Admixture results (&lt; 1 min)</li>
</ol>
<p>All done in R locally on your machine</p>
<p><br />
<br />
</p>
</div>
<div id="other-info" class="section level3">
<h3>Other info</h3>
<ul>
<li>If you are on a PC, directory paths need to be separated by back-slashes instead of forward-slashes</li>
<li>Try not to resize windows/display panels while R is busy calculating/plotting - this tends to cause problems</li>
</ul>
<p><br />
<br />
</p>
</div>
<div id="initial-setup-and-package-install" class="section level3">
<h3>1. Initial setup and package install</h3>
<p>The following block of code will install all required R packages for this tutorial, you can copy and paste this in your R Studio and run it. All of this code works with the latest R version (4.1.0), and possibly older ones (e.g. 3.5.0), but I cannot guarantee it will work, so best to have R v.4.1.0.</p>
<pre class="r"><code>rm(list = ls(all = TRUE))

# install.packages(adegenet)
library(adegenet)</code></pre>
<p><br />
<br />
</p>
</div>
<div id="load-data-and-convert-to-genind-format-for-adegenet" class="section level3">
<h3>2. Load data and convert to genind format for Adegenet</h3>
<p>Load the .dat input file that we will analyse. I’ve already converted this for you into a .dat (fstat format) file. Easiest way to do this is using PGDSpider (<a href="http://www.cmpg.unibe.ch/software/PGDSpider/" class="uri">http://www.cmpg.unibe.ch/software/PGDSpider/</a>).</p>
<p>Again, this is a fancy file structure which is basically just a matrix of individuals and genotypes across all SNPs…</p>
<pre class="r"><code># Read in .dat file (fstat) format needed by Adegenet package for running DAPC
data &lt;- read.fstat(&quot;./data/L_flavomaculatus/Outputs/DAPC/Lflavomaculatus.dat&quot;)
## 
##  Converting data from a FSTAT .dat file to a genind object... 
## 
## 
## ...done.</code></pre>
<p>We can check the size of the file with the length argument (should be 59 individuals)</p>
<pre class="r"><code>length(indNames(data))
## [1] 59</code></pre>
<p><br />
<br />
</p>
</div>
<div id="run-the-dapc-and-plot-bic-scores" class="section level3">
<h3>3. Run the DAPC and plot BIC scores</h3>
<p>We then run the DAPC itself using find.clusters, max.n.clusters can be up to any number (&lt;indiviudals), but let’s set it to 10 here to keep in line with the Admixture and sNMF analyses. Take all possible PCAs (so 60, and press [enter]</p>
<pre class="r"><code>grp &lt;- find.clusters(data, max.n.clust = 10, n.pca = 59, choose.n.clust = FALSE,
    criterion = &quot;min&quot;)
# the last 3 arguments here have automated this, highly recommended to remove
# these and do it interactively in R by looking at the resulting plots Advice:
# number of PCs to retain - here say (number of individuals - 1 to maximise the
# information initially) number of clusters - here select a sensible lowest
# BIC. In our case we automatically select 4 (min) because the elbow is very
# clear, in other datasets this my not be the case...</code></pre>
<p>It will ask you to choose the number of clusters. As you can see, K=4 again is best. We can look at some summary stuff here which is useful (BIC scores, best K, Population assignment per individual, number of Inds in each population cluster:</p>
<pre class="r"><code>head(grp$Kstat)  # BIC scores per k
grp$stat  # Best k based on BIC
head(grp$grp, 59)  # Group membership (based on your defined number of clusters)
grp$size  # Group size per k cluster

x &lt;- as.data.frame(grp$Kstat)
png(filename = &quot;./outputs/DAPC_Results_BIC_scores.png&quot;, type = &quot;quartz&quot;, height = 8,
    width = 8, units = &quot;in&quot;, res = 300)
plot(x$`grp$Kstat`, pch = 19, col = &quot;black&quot;, cex = 0.25, xlab = &quot;k clusters&quot;, ylab = &quot;Bayesian Information Criterion&quot;,
    main = &quot;BIC across k&quot;)
lines(x$`grp$Kstat`, lty = 1, lwd = 1.5, col = &quot;steelblue2&quot;)
dev.off()
##      K=1      K=2      K=3      K=4      K=5      K=6 
## 376.6955 373.2774 371.9943 371.4241 372.7660 374.8831 
##      K=4 
## 371.4241 
##  1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19 20 21 22 23 24 25 26 
##  4  4  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  1  2  2  4  4  4 
## 27 28 29 30 31 32 33 34 35 36 37 38 39 40 41 42 43 44 45 46 47 48 49 50 51 52 
##  4  4  4  4  4  4  4  4  4  4  4  4  4  4  4  4  2  4  4  4  4  4  4  4  4  3 
## 53 54 55 56 57 58 59 
##  3  3  3  3  3  3  3 
## Levels: 1 2 3 4
## [1] 19  3  8 29
## quartz_off_screen 
##                 2</code></pre>
<p>Now let’s do the DAPC. Run the below code. Thsi time choose a lower number of retained PCAs, ideally where the curve starts to become less (e.g. 4 or 5 in this case). Press [enter] and then choose the number of discriminant functions to retain based on the eigenvalues. In this case you want to take all 3, but depends on the information content (F-statistic).</p>
<pre class="r"><code># now do the DAPC
dapc1 &lt;- dapc(data, grp$grp, n.pca = 5, n.da = 3)
# same as above, this is usually interactive - the last 2 arguments here have
# automated this, highly recommended to remove these and do it interactively in
# R by looking at the resulting plots Advice: number of PCs to retain - now we
# can be more selective, pick a number where the initial curve of the plot
# starts to slow (in our case 4 or 5) number of discriminant functions to
# retain - always more than one - general rule is to take anything which is
# quite a lot higher than 0, so in this case say 3. If our eigenvalues here
# showed two big bars and then a very small one we would take 2. More details
# on how to interpret and select are found here:
# https://adegenet.r-forge.r-project.org/files/tutorial-dapc.pdf</code></pre>
<p><br />
<br />
</p>
</div>
<div id="make-dapc-plot-of-results-1-min" class="section level3">
<h3>4. Make DAPC plot of results (&lt; 1 min)</h3>
<p>Set some colours and then plot</p>
<pre class="r"><code>myCol &lt;- c(&quot;blue&quot;, &quot;turquoise&quot;, &quot;lightblue&quot;, &quot;darkblue&quot;)
png(&quot;./outputs/DAPC_Lflavomaculatus.png&quot;, width = 500, height = 500)
scatter(dapc1, scree.da = TRUE, scree.pca = TRUE, bg = &quot;white&quot;, pch = 20, cell = 0,
    cstar = 0, col = myCol, solid = 1, cex = 1.5, clab = 0, leg = FALSE, txt.leg = paste(&quot;Cluster&quot;,
        1:4))
dev.off()
## quartz_off_screen 
##                 2</code></pre>
<p><br />
<br />
</p>
</div>
<div id="on-your-own" class="section level3">
<h3>On your own…</h3>
<p>Mess around with PCAs retained and number of clusters to see how it affects the BIC values, curves and population assignment. It can be really tricky without god knowledge of true K, especiallly in not well structured populations</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
