<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tutorial: sNMF</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ddRAD-seq_workshop</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00_setup.html">00 - Setting up the data and scripts for analysis</a>
    </li>
    <li>
      <a href="01_download_data.html">01 - Downloading data from ENA</a>
    </li>
    <li>
      <a href="02_process_radtags.html">02 - Demultiplexing sequencing data (process_radtags)</a>
    </li>
    <li>
      <a href="03_denovo_map.html">03 - Running Stacks 2 (denovo_map)</a>
    </li>
    <li>
      <a href="04_Admixture.html">04 - Population structure (Admixture)</a>
    </li>
    <li>
      <a href="05_sNMF.html">05 - Population structure (sNMF)</a>
    </li>
    <li>
      <a href="06_DAPC.html">06 - Population structure (DAPC)</a>
    </li>
    <li>
      <a href="07_RAxML.html">07 - Phylogenetic relationships (RAxML)</a>
    </li>
    <li>
      <a href="08_F-stats.html">08 - Genetic Diversity (F-stats)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/cd-barratt/ddRAD-seq_workshop/">GitHub</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Tutorial: sNMF</h1>

</div>


<p><br />
<br />
</p>
<div id="timings" class="section level3">
<h3>Timings</h3>
<ol style="list-style-type: decimal">
<li>Load data (&lt; 1 min)</li>
<li>Run sNMF (3-4 mins)</li>
<li>Plot CV scores (&lt; 1 min)</li>
<li>Make barplot of Admixture results (&lt; 1 min)</li>
</ol>
<p>All done in R locally on your machine</p>
<p><br />
<br />
</p>
</div>
<div id="objectives" class="section level3">
<h3>Objectives</h3>
<p>Again, using the same <a href="https://amphibiaweb.org/species/3649"><em>Leptopelis flavomaculatus</em></a> data we will try a slightly different but very similar approach to look at population structure. Instead of Admixture we will use <a href="http://membres-timc.imag.fr/Olivier.Francois/LEA/tutorial.htm">sNMF</a> from the LEA R package this time</p>
<p><br />
<br />
</p>
</div>
<div id="load-data" class="section level3">
<h3>1. Load data</h3>
<p>Load the .ped input file that we will analyse. This is a matrix (rows = number of individual samples, columns = genotypes at each locus). The first 6 columns are mandatory (family, individual, maternal/paternal IDs, Sex abd Phenotype). We actually only have the individual IDs so the others are defaulted out. You can read more about this PLNK, format <a href="https://zzz.bwh.harvard.edu/plink/data.shtml">here</a></p>
<pre class="r"><code># Read in ped file, convert to .geno format needed by LEA package for sNMF
library(LEA)

data = read.table(&quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/data/Exercises_4-8/sNMF/Lflav.ped&quot;)
output = ped2geno(&quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/data/Exercises_4-8/sNMF/Lflav.ped&quot;)
geno &lt;- read.geno(&quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/data/Exercises_4-8/sNMF/Lflav.geno&quot;)</code></pre>
<p><br />
<br />
</p>
</div>
<div id="run-snmf" class="section level3">
<h3>2. Run sNMF</h3>
<p>We can give sNMF a range of K to test, so let’s follow 1:10 like we did in Admixture so that we have comparable results. Here we set the ploidy to 2 (diploid organism, having 5 repetitions of each K value to test, so 50 total reps). We set entropy=T and project=‘new’). There are a bunch of things you can tweak (see here: <a href="http://membres-timc.imag.fr/Olivier.Francois/snmf/files/note.pdf" class="uri">http://membres-timc.imag.fr/Olivier.Francois/snmf/files/note.pdf</a>) but let’s keep this simple. Current settings here will take around 3-4 mins to complete</p>
<p>Block to update:</p>
<pre class="r"><code># Run snmf to look at structure
k_range &lt;- 1:10
geno.snmf &lt;- snmf((&quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/data/Exercises_4-8/sNMF/Lflav.geno&quot;),
    K = k_range, entropy = T, ploidy = 2, rep = 5, project = &quot;new&quot;)</code></pre>
<p><br />
<br />
</p>
</div>
<div id="plot-cv-scores" class="section level3">
<h3>3. Plot CV scores</h3>
<p>We will first set the plot margins and then set an output file (png) with specific sizes and resolutions Then we plot the results (first points, then a line to join the points) and save the file</p>
<pre class="r"><code># plot CV scores
sumpr2 &lt;- summary(geno.snmf)$crossEntropy
sumpr &lt;- sumpr2[2, ]

par(mar = c(2, 2, 2, 2))
png(filename = &quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/outputs/Exercise_5/sNMF_Results_CV_scores.png&quot;,
    type = &quot;quartz&quot;, height = 8, width = 8, units = &quot;in&quot;, res = 300)
plot(sumpr, lty = 2, lwd = 1.5, col = &quot;black&quot;, pch = 19, cex = 0.5, xlab = &quot;k clusters&quot;,
    ylab = &quot;CV score&quot;, main = &quot;Cross validation scores across k (sNMF)&quot;)
lines(sumpr, lwd = 2, col = &quot;steelblue2&quot;)
dev.off()</code></pre>
<p><br />
<br />
</p>
</div>
<div id="make-barplot-of-snmf-results" class="section level3">
<h3>4. Make barplot of sNMF results</h3>
<p>We’ll first do an automated thing to get the best value of K (which K value and which run of the 5 you made)</p>
<pre class="r"><code># Choose K with lowest mean cross entropy
K &lt;- c(1:10)[which.min(sumpr2[2, ])]  # K=4
# Get the cross-entropy of each run for K
ce &lt;- cross.entropy(geno.snmf, K = K)
# Select the run with the lowest cross-entropy
best &lt;- which.min(ce)
best</code></pre>
<p>Next we will take the q values from the sNMF results for the selected K value and run (analagous to the Admixture Lflav.4.Q file). It is a matrix of 59 rows (one per individual) and 4 columns (i.e. K). The probability of membership to each population cluster is shown for each individual</p>
<pre class="r"><code>Q_vals &lt;- Q(geno.snmf, K = K, run = best)
sample_names &lt;- data[, 2]
pop_assignment &lt;- as.data.frame(cbind(sample_names, Q_vals))

ord = pop_assignment[order(pop_assignment$V4, pop_assignment$V1, pop_assignment$V2,
    pop_assignment$V3), ]
rownames(ord) &lt;- ord$sample_names
ord &lt;- ord[, 2:5]

par(mar = c(10, 4, 1, 1))
bp = barplot(t(as.matrix(ord)), space = (0.04), col = c(&quot;blue&quot;, &quot;turquoise&quot;, &quot;darkblue&quot;,
    &quot;lightblue&quot;), xlab = &quot;&quot;, ylab = &quot;Ancestry&quot;, border = 1, las = 2, cex.names = 0.9)</code></pre>
<p>The above plot is not so clean, very messy and not very pretty. Let’s reorder the output file so that it matches the order of individuals in the Admixture analyses. It can then be plotted for direct and easy comparison.</p>
<pre class="r"><code># match ord_labels to the pop_assignment
ord_labels &lt;- read.csv(&quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/data/Exercises_4-8/sNMF/ord_labels_sNMF.csv&quot;)
ord_labels_sample_name_only &lt;- as.data.frame(ord_labels)
colnames(ord_labels_sample_name_only) &lt;- &quot;sample_name_only&quot;
ord_labels_sample_name_only &lt;- substr(ord_labels_sample_name_only$sample_name_only,
    1, 5)
ord_labels_sample_name_only &lt;- as.data.frame(ord_labels_sample_name_only)
ord_labels &lt;- cbind(ord_labels, ord_labels_sample_name_only)

# now sort pop_assignment based on order in ord_labels$sample_name_only

sorted_pop_assignment &lt;- pop_assignment[order(match(pop_assignment[, 1], ord_labels_sample_name_only[,
    1])), ]
row.names(sorted_pop_assignment) &lt;- sorted_pop_assignment$sample_names
sorted_pop_assignment &lt;- sorted_pop_assignment[, 2:5]

# plot the sNMF barplot for comparison with Admixture
bp = barplot(t(as.matrix(sorted_pop_assignment)), space = (0.04), col = c(&quot;blue&quot;,
    &quot;lightblue&quot;, &quot;turquoise&quot;, &quot;darkblue&quot;), xlab = &quot;&quot;, ylab = &quot;Ancestry&quot;, border = 1,
    las = 2, cex.names = 0.9)

# save
png(filename = &quot;/Users/cb76kecu/Desktop/ddRAD-seq_workshop/outputs/Exercise_5/sNMF_Results_Barplot.png&quot;,
    type = &quot;quartz&quot;, height = 8, width = 30, units = &quot;in&quot;, res = 300)
bp = barplot(t(as.matrix(sorted_pop_assignment)), space = (0.04), col = c(&quot;blue&quot;,
    &quot;lightblue&quot;, &quot;turquoise&quot;, &quot;darkblue&quot;), xlab = &quot;&quot;, ylab = &quot;Ancestry&quot;, border = 1,
    las = 2, cex.names = 0.9)
dev.off()
</code></pre>
<p><br />
<br />
</p>
</div>
<div id="on-your-own" class="section level3">
<h3>On your own…</h3>
<p>Like with Admixture, If you want, try plotting k=1 - k=5, feel free to change colour schemes and the order of the individuals. Again, you can edit the number of reps from 5 to 10 or higher for example. If you have your own data you can try and use this script to plot your own outputs</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
