<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Tutorial: RAxML</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-inverse  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">ddRAD-seq_workshop</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fa fa-home"></span>
     
    Home
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Tutorials
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="00_setup.html">00 - Setting up the data and scripts for analysis</a>
    </li>
    <li>
      <a href="01_download_data.html">01 - Downloading data from ENA</a>
    </li>
    <li>
      <a href="02_process_radtags.html">02 - Demultiplexing sequencing data (process_radtags)</a>
    </li>
    <li>
      <a href="03_denovo_map.html">03 - Running Stacks 2 (denovo_map)</a>
    </li>
    <li>
      <a href="04_Admixture.html">04 - Population structure (Admixture)</a>
    </li>
    <li>
      <a href="05_sNMF.html">05 - Population structure (sNMF)</a>
    </li>
    <li>
      <a href="06_DAPC.html">06 - Population structure (DAPC)</a>
    </li>
    <li>
      <a href="07_RAxML.html">07 - Phylogenetic relationships (RAxML)</a>
    </li>
    <li>
      <a href="08_F-stats.html">08 - Genetic Diversity (F-stats)</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/cd-barratt/ddRAD-seq_workshop/">GitHub</a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Tutorial: RAxML</h1>

</div>


<p><br />
<br />
</p>
<div id="timings" class="section level3">
<h3>Timings</h3>
<ol style="list-style-type: decimal">
<li>Run RAxML (ca. 5 min for 5 bootstraps and full MLE search, higher numbers of bootstraps will run exponentially longer)</li>
</ol>
<p>All done in EVE cluster</p>
<p><br />
<br />
</p>
</div>
<div id="objectives" class="section level3">
<h3>Objectives</h3>
<p>We’ve looked at 3 methods for looking at population structure. We can also confirm this information on population structure from a phylogeny, by looking at how the individuals and populations cluster within a tree. More interestingly we can look at the relationships between populations (something that is not particularly clear with Admixture, sNMF and DAPC analyses)</p>
<p> </p>
<p>For this exercise we will use <a href="https://cme.h-its.org/exelixis/web/software/raxml/">RAxML</a> on the EVE cluster to investigate phylogenetic relationships. Note that we will not do any phylogenetic dating here, for that you could use a variety of other things, for example <a href="https://beast.community/">BEAST</a>.</p>
<p><br />
<br />
</p>
</div>
<div id="run-raxml" class="section level3">
<h3>1. Run RAxML</h3>
<p>#This is all done in the EVE cluster, it will take some time - we can reduce the number of bootstrap replicates to 5. You can build your own script or use the supplied one that you can edit (07_RAxML.sh)</p>
<pre class="r"><code>#!/bin/bash                                                                                                                            

#SBATCH --job-name=RAxML
#SBATCH --mail-user=christopher_david.barratt@uni-leipzig.de
#SBATCH --mail-type=BEGIN,END,FAIL,TIME_LIMIT
#SBATCH --output=/work/$USER/ddRAD-seq_workshop/job_logs/%x-%j.log
#SBATCH --cpus-per-task=1 
#SBATCH --mem-per-cpu=40G
#SBATCH --time=12:00:00

#load environment
module purge
module load Anaconda3
source activate /global/apps/raxml_8.2.9

cd /work/$USER/ddRAD-seq_workshop/data/Exercises_4-8/RAxML/

# 10 bootstraps


raxmlHPC -f a -m GTRCAT -p 12345 -x 12345 -# 5 -s  /work/$USER/ddRAD-seq_workshop/data/Exercises_4-8/RAxML/Lflav.phy ­­asc­corr=lewis -n bootstrapped 
raxmlHPC -m GTRCAT -p 12345 -f b -t RAxML_bestTree.bootstrapped -z RAxML_bootstrap.bootstrapped -n FINAL_bootstrapped.tre

# move output files to outputs directory
mv /work/$USER/ddRAD-seq_workshop/data/Exercises_4-8/RAxML/RAxML*.* /work/$USER/ddRAD-seq_workshop/outputs/Exercise_7/</code></pre>
<p><br />
<br />
</p>
</div>
<div id="visualise-the-tree" class="section level3">
<h3>Visualise the tree</h3>
<p>First we can run rsync again to pull in the files from RAxML into our desktop directory:</p>
<p><br />
</p>
<pre class="r"><code># transfer from the cluster
rsync -avhP -e &#39;ssh -p 8022&#39; \
  barratt@localhost:/work/$USER/ddRAD-seq_workshop \
  /Users/cb76kecu/Desktop</code></pre>
<p><br />
</p>
<p>Then…</p>
<ul>
<li><p>Download and install <a href="https://www.softpedia.com/dyn-postdownload.php/e272921fefc6c686b1fa2b06df64a860/62a88e68/2a33f/4/1">Figtree</a> on your computer.</p></li>
<li><p>Rsync the files from the EVE cluster back to your own computer again</p></li>
<li><p>Open Figtree and open the “RAxML_bipartitions.FINAL_bootstrapped.tre” file</p></li>
<li><p>A window will come up, just click ok (it’s about how to visualise the bootstrap values on nodes)</p></li>
<li><p>Your tree will appear, let’s visualise it better…</p></li>
<li><p>Open the “Trees” panel on the left, tick the “Root tree” box and then select “Rooting: Midpoint”</p></li>
<li><p>Go to the main menu and say “File &gt; Import Annotations”. When the dialog box opens select the <code>ddRAD-seq_workshop/data/Exercises_4-8/RAxML/RAD_IDs.txt</code> file</p></li>
<li><p>Go back to Figtree and then open the “Tip Labels” panel, select “Display: NEW NAME”</p></li>
<li><p>You’ll see that the tip labels have now been replaced and you can compare them easily with the Admixture, sNMF and DAPC results</p></li>
</ul>
<p><br />
<br />
</p>
</div>
<div id="on-your-own" class="section level3">
<h3>On your own…</h3>
<p>You can rerun RAxML with different parameter settings, for example by changing the number of bootstraps by changing the number in the first line to -# 10 instead of -# 5 (100 recommended for a full analysis, but this takes time). You could also look at different models of evolution instead of GTRCAT, or try a different ascertainment bias correction (not lewis)</p>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
